src/cryptography/x509/base.py

```diff
@@ -345,7 +346,11 @@ def sign(
                 raise TypeError("Padding is only supported for RSA keys")
 
```

Comment:
I see that we check `rsa_padding` against the key type here. Surely we want to do the same for ecdsa_deterministic? (Which probably means making the default `None`, so we can error on explicit `=False`)?